src/e2e-test/java/io/cdap/plugin/pubsub/stepsdesign/PubSubConnector.java

```diff
@@ -0,0 +1,401 @@
+  }
+
+  @Then("Run the Pipeline in Runtime")
+  public void runThePipelineInRuntime() throws InterruptedException {
+    CdfPipelineRunAction.runClick();
+  }
+
+  @Then("Wait till pipeline is in running state")
+  public void waitTillPipelineIsInRunningState() throws InterruptedException {
+    WebDriverWait wait = new WebDriverWait(SeleniumDriver.getDriver(), 300);
+    wait.until(ExpectedConditions.or(
+      ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@data-cy='Succeeded']")),
+      ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@data-cy='Failed']"))));
+  }
+
+  @Then("Verify the pipeline status is {string}")
+  public void verifyThePipelineStatusIs(String status) {
+    boolean webelement = false;
+    webelement = SeleniumHelper.verifyElementPresent("//*[@data-cy='" + status + "']");
+    assertTrue(webelement);
+  }
+
+  @Then("Open Logs")
+  public void openLogs() {
+    CdfPipelineRunAction.logsClick();
+  }
+
+  @Then("Enter the PubSub property with blank property {string}")
+  public void enterThePubSubPropertyWithBlankProperty(String property) throws IOException {
+    if (property.equalsIgnoreCase("referenceName")) {
+      PubSubActions.enterPubsubTopic("dummyTopic");
+    } else if (property.equalsIgnoreCase("topic")) {
+      PubSubActions.enterPubSubReferenceName();
+    }
+  }
+
+  @Then("Validate mandatory property error for {string}")
+  public void validateMandatoryPropertyErrorFor(String property) {
+    CdfStudioActions.clickValidateButton();
+    SeleniumHelper.waitElementIsVisible(CdfStudioLocators.validateButton);
+    E2ETestUtils.validateMandatoryPropertyError(property);
+  }
+
+  @Then("Enter the PubSub advanced properties with incorrect property {string}")
+  public void enterThePubSubAdvancedPropertiesWithIncorrectProperty(String property) {
+    if (property.equalsIgnoreCase("messageCountBatchSize")) {
+      SeleniumHelper.replaceElementValue(PubSubLocators.maximumBatchcount,
+                                         E2ETestUtils.pluginProp("pubSubStringValue"));
+    } else if (property.equalsIgnoreCase("requestThresholdKB")) {
+      SeleniumHelper.replaceElementValue(PubSubLocators.maximumBatchSize,
+                                         E2ETestUtils.pluginProp("pubSubStringValue"));
+    } else if (property.equalsIgnoreCase("publishDelayThresholdMillis")) {
+      SeleniumHelper.replaceElementValue(PubSubLocators.publishDelayThreshold,
+                                         E2ETestUtils.pluginProp("pubSubStringValue"));
+    } else if (property.equalsIgnoreCase("retryTimeoutSeconds")) {
+      SeleniumHelper.replaceElementValue(PubSubLocators.retryTimeout,
+                                         E2ETestUtils.pluginProp("pubSubStringValue"));
+    } else if (property.equalsIgnoreCase("errorThreshold")) {
+      SeleniumHelper.replaceElementValue(PubSubLocators.errorThreshold,
+                                         E2ETestUtils.pluginProp("pubSubStringValue"));
+    }
```

Comment:
fail test if property is not one of the above values

Reply:
added else block.